src/main/kotlin/no/nav/familie/ef/sak/behandling/oppgaveforopprettelse/OppgaverForOpprettelseService.kt

```diff
@@ -0,0 +1,65 @@
+package no.nav.familie.ef.sak.behandling.oppgaveforopprettelse
+
+import no.nav.familie.ef.sak.behandling.BehandlingService
+import no.nav.familie.ef.sak.behandling.domain.Behandling
+import no.nav.familie.ef.sak.behandling.domain.BehandlingType
+import no.nav.familie.ef.sak.infrastruktur.exception.feilHvisIkke
+import no.nav.familie.ef.sak.tilkjentytelse.TilkjentYtelseService
+import no.nav.familie.ef.sak.tilkjentytelse.domain.TilkjentYtelse
+import no.nav.familie.kontrakter.ef.iverksett.OppgaveForOpprettelseType
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.util.UUID
+
+@Service
+class OppgaverForOpprettelseService(
+    private val oppgaverForOpprettelseRepository: OppgaverForOpprettelseRepository,
+    private val behandlingService: BehandlingService,
+    private val tilkjentYtelseService: TilkjentYtelseService,
+) {
+
+    @Transactional
+    fun opprettEllerErstattFremleggsoppgave(behandlingId: UUID, nyeOppgaver: List<OppgaveForOpprettelseType>) {
+        val oppgavetyperSomKanOpprettes = hentOppgavetyperSomKanOpprettes(behandlingId)
+        feilHvisIkke(oppgavetyperSomKanOpprettes.containsAll(nyeOppgaver)) {
+            "behandlingId=$behandlingId prøver å opprette $nyeOppgaver $oppgavetyperSomKanOpprettes"
+        }
+        when (this.oppgaverForOpprettelseRepository.existsById(behandlingId)) {
+            true -> this.oppgaverForOpprettelseRepository.update(OppgaverForOpprettelse(behandlingId, nyeOppgaver))
+            false -> this.oppgaverForOpprettelseRepository.insert(OppgaverForOpprettelse(behandlingId, nyeOppgaver))
```

Comment:
Vil vi lage en rad for alle de som sender inn en tom liste?

Reply:
Eller bør vi slette innholdet da?

Reply:
En tom liste betyr at det er gjort et valg om å ikke opprette oppgave

Reply:
> En tom liste betyr at det er gjort et valg om å ikke opprette oppgave. Hvis vi ikke legger inn innhold da, så vil ikke valget som er foretatt kunne hentes opp ?

Ikke nødvendigvis? 
Hvis ingen oppgaver kan opprettes så får du også inn tom liste her. 
Mulig jeg ikke helt skjønner forskjellen på tom liste og ingen verdi - blir det markert forskjellig i frontend?

Reply:
Hvis den skal "default" være på - betyr det at vi kanskje burde hatt en "ryddejobb" etterhvert og slettet de "tomme" radene i databasen?

Reply:
Ja, dette er en bug, bra observert. Det skal ikke komme noe innslag i tabellen når det ikke kan opprettes oppgave / ikke har blitt tatt noe valg.

Reply:
Sjekker nå om det kan opprettes en oppgave før vi legger inn noe